MIPS/mip-0005-mip-upgrade.md

```diff
@@ -0,0 +1,315 @@
+- Follow the instructions provided in the GitHub/MIPs documentation.
+- Collect and incorporate feedback from the community.
+- Document alternative opinions, the options considered and other discussed topics.
+- Champion the MIP through each stage of the process.
+- Designate, coordinate and oversee a Dev Team for the MIP implementation.
+
+**Dev Team:** core development team, contributors and any other active participants that are making updates, parameter changes, and upgrades to the protocol itself. 
+
+Responsibilities:
+- Review MIP’s implementation plan.
+- Execute the implementation plan of approved MIPs.
+
+Functions:
+- Assess the feasibility of the MIP’s implementation plan and provide feedback to the MIP Author.
+- Ensure correct execution of the implementation plan in a timely manner.
+- Communicate and explain any delays and other important decisions about the implementation.
+
+#### Introducing New Roles and Responsibilities:
+
+**MIP Facilitator:** Elected community member with a deep understanding of Mina Protocol’s governance to oversee that the MIP process is followed correctly.
+
+Responsibilities:
+- Ensure a smooth progression of MIPs through the process.
+- Facilitate cooperation and collaboration between all participants and ensure they follow the rules.
+- Assist and support MIP Authors as necessary.
+
+Functions:
+- Accept or reject a MIP submission based on the established approval criteria.
+- Accept or reject a status change request ensuring that each step of the process is completed correctly.
+- Schedule and facilitate meetings between MIP Author and MIP Review Group, and other stakeholders.
+- Produce and share meeting minutes under the Chatham House Rule.
+- Identify key stakeholders affected by the MIP.
+- Coordinate communications with the Mina Foundation.
+
+**MIP Reviewer:** Elected community members who have expertise to review a MIP in a specific category.
+
+Responsibilities:
+- Review the assigned MIP.
+- Provide feedback to the Author to improve the MIP.
+
+Functions:
+- Identify risks, benefits and tradeoffs of the proposed changes.
+- Assess the risk of a MIP to the long-term health of the protocol.
+- Publish a Risk Assessment with a rationale.
+- Join the meetings scheduled by the MIP Facilitator.
+
+#### New Role Frameworks
+
+To ensure accountability and transparency for MIP Facilitators and Reviewers, we propose developing frameworks that clearly explain the processes for role selection, assignment, removal, resignation, and compensation. 
+
+These frameworks will share common principles, including:
+
+- Any community member can propose themselves as a candidate for MIP Facilitator or MIP Reviewer by submitting a new MIP.
+- Any community member can propose the removal of an existing MIP Facilitator or MIP Reviewer by submitting a new MIP.
+- Any community member can vote to approve or reject the selection or removal of MIP Facilitators and MIP Reviewers.
```

Comment:
Will there be a full on chain vote every single time? Have we done an analysis on the expected transaction throughput a MIP vote has on the network? If we do too many open votes at once, we could run into congestion issues on the chain.

Reply:
It's an important point! On-chain voting is indeed central to ensuring a transparent and democratic process. To address potential risks of network congestion and voter fatigue, this MIP Upgrade highlights the critical role of MIP Facilitators in managing proposal throughput. Facilitators can coordinate the timing of proposals and group multiple votes into a single transaction, minimizing individual voting events while upholding the integrity of the governance process.

Although a formal analysis of transaction throughput for MIP votes hasn’t been conducted yet, this is a valuable area to explore as the governance framework evolves. I appreciate your input and welcome any further suggestions on optimizing efficiency and inclusivity in this process.